static/src/javascripts/projects/common/modules/atoms/story-questions.js

```diff
@@ -0,0 +1,260 @@
+// @flow
+
+import mediator from 'lib/mediator';
+import { getViewport } from 'lib/detect';
+import $ from 'lib/$';
+import fetch from 'lib/fetch';
+import bean from 'bean';
+import config from 'lib/config';
+import ophan from 'ophan/ng';
+import { getUserFromApi } from 'common/modules/identity/api';
+import fastdom from 'fastdom';
+import { submitComponentEvent } from 'common/modules/commercial/acquisitions-ophan';
+
+const sendOldStyleInteraction = (
+    atomId: string,
+    component: string,
+    value: string
+): void => {
+    ophan.record({
+        atomId,
+        component,
+        value,
+    });
+};
+
+const sendNewStyleInteraction = (
+    atomId: string,
+    action: string,
+    value?: string,
+    eventId?: string
+): void => {
+    const event: Object = {
+        component: {
+            componentType: 'READERS_QUESTIONS_ATOM',
+            id: atomId,
+        },
+        action,
+    };
+
+    if (value) {
+        event.value = value;
+    }
+
+    if (eventId) {
+        event.id = eventId;
+    }
+
+    submitComponentEvent(event);
+};
+
+const askQuestion = (
+    event: Event,
+    isEmailSubmissionReady: boolean | string
+): void => {
+    event.preventDefault();
+
+    const currentTarget = ((event.target: any): HTMLElement);
+    const askQuestionBtn = currentTarget.querySelector(
+        '.user__question-upvote'
+    );
+    const atomIdElement = $('.js-storyquestion-atom-id');
+
+    if (askQuestionBtn && atomIdElement) {
+        const questionId = askQuestionBtn.dataset.questionId;
+        const atomId = atomIdElement.attr('id');
+
+        const question = ((document.querySelector(
+            `meta[name=js-notranslate-${questionId}]`
+        ): any): HTMLMetaElement);
+
+        if (question) {
+            const questionText = question.content;
+
+            if (questionText && atomId) {
+                if (askQuestionBtn) {
+                    askQuestionBtn.classList.add('is-hidden');
+                }
+
+                if (isEmailSubmissionReady === 'true') {
+                    /* I had to make this a one-liner, because otherwise I
+                       couldn't find a way to tell flow to ignore the next
+                       line */
+                    /* eslint-disable prettier/prettier */
+                    // $FlowFixMe
+                    const signupForm = document.forms[`js-storyquestion-email-signup-form-${questionId}`];
+                    /* eslint-enable */
```

Comment:
Ack, what a pain! We could potentially grab the form element by ID, rather than using `document.forms` (source: [this comment](https://github.com/guardian/frontend/pull/16976#discussion_r119835295) and [this line of HTML](https://github.com/guardian/frontend/blob/master/common/app/views/fragments/atoms/storyquestions.scala.html#L67)). 

@regiskuckaertz may have something to say about this, though!

Reply:
Can we use [`document.forms.namedItem(...)`](https://github.com/facebook/flow/blob/master/lib/dom.js#L513)?

Reply:
Alternatively `document.forms.namedItem('...')` could work too. It's one of those instances where Flow should not go against a perfectly legitimate HTML API.

Reply:
<img width="586" alt="screen shot 2017-10-04 at 15 48 31" src="https://user-images.githubusercontent.com/2244375/31182339-81243a9a-a91b-11e7-8e49-ddd5bd1a5f9e.png">

Could do, but we'll have to think about something better for the future...

Reply:
Where is this coming from? IE does support `document.forms`, they may have invented `HTMLCollection`

Reply:
From [mdn](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormControlsCollection). Will have a look into that ...

Reply:
That one is to iterate over elements inside a form, not over forms inside a document. But anyway, IE should have support too 😮